homeassistant/components/switch/qwikswitch.py

```diff
@@ -0,0 +1,37 @@
+"""
+Support for Qwikswitch Relays as HA Switches.
+
+See the main component for more info
+"""
+import logging
+from homeassistant.components.qwikswitch import QSToggleEntity
+from homeassistant.components.switch import SwitchDevice
+
+DEPENDENCIES = ['qwikswitch']
+
+
+class QSSwitch(QSToggleEntity, SwitchDevice):
+    """Switch based on a Qwikswitch relay module."""
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Store add_devices for the 'switch' components."""
+    if discovery_info is None or 'qsusb' not in discovery_info:
+        _LOGGER.error('Configure main Qwikswitch component')
+        return False
+
+    _LOGGER.info('Qwikswitch light setup_platform called')
+    qsusb = discovery_info['qsusb']
```

Comment:
All data sent over the event bus must be serializable. `qsusb` is not serializable. Please do a similar approach to Z-Wave: the main component will have a global variable `NETWORK` that the platforms refer to.